cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -172,7 +202,10 @@ func (s *Server) Run() {
 						return
 					}
 					metricResolveRevisionDuration.WithLabelValues("true").Observe(time.Since(start).Seconds())
-					queue.AddOrUpdate(name, commit)
+					added := queue.AddOrUpdate(name, commit)
+					if added {
+						metricQueueLen.WithLabelValues(codeHostFromName(name)).Add(1.0)
```

Comment:
This is prone to logic failures where `added` is incorrect, or even if `queue.AddOrUpdate` for some reason does/does not actually enqueue the update.

I think a better approach would be to set `metricQueueLen` directly to the value of `queue.Len()` because then you _know_ it's actually the real underlying queue length.

Reply:
If you're worried about the cost of locking (I think it's negligible here), have `queue.AddOrUpdate` return the length instead of an added boolean.

Reply:
true, you're absolutely right. i had it like that. but unfortunately, it won't do it by code host. so i could give up breaking it down by code host and then simplify this as you suggested or i have to leave this as is and hope it's right.

Reply:
Oh I missed this was by code host, LGTM then.